### PR TITLE
0608-grammars

### PR DESCRIPTION
**Pull request by Luluyu2022**
"uielement" to "UiElement"
"ex" to "eg"
some grammars have been updated

### CODE REVIEW

File: doc/api/python/exceptions/elementcannotfounderror.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 - extends: [BaseError](./doc/api/python/exceptions/baseerror.md)
 
-**ElementCanNotFoundError is raised when the loactor can be found in locator store, but the returned ui element can not be found.**
+**ElementCanNotFoundError is raised when the loactor can be found in locator store, but the returned Uielements can not be found.**
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/elementcannotfounderror.md:5]: Uielements -> UI elements

File: doc/api/python/exceptions/exceptions.md [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
+| [ExtensionOperationError](./doc/api/python/exceptions/extensionoperationerror.md) | ExtensionOperationError is raised when the specified extension with oparetion "install", "update", or "uninstall" failed.|
+| [NotSupportedError](./doc/api/python/exceptions/notsupportederror.md) | NotSupportedError is raised when an invoked method is not supported, or when an attempt to read, seek, or write to a stream that does not support the invoked functionality.|
+| [NotSupportedOperationError](./doc/api/python/exceptions/notsupportedoperationerror.md) | NotSupportedOperationError is raised when an invoked method, its automation technology or operation is not supported for the target. Eg: can not set text for an image UiElement. |
+| [NotSupportedOperationOptionError](./doc/api/python/exceptions/notsupportedoperationoptionerror)   | NotSupportedOperationOptionError is raised when an invoked method is supported for the target, but the specified option value is not supported. Eg: can not clear text for desktop Uielement with clear method option set to `ControlClearValue`|
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/exceptions.md:20]: UiElement

File: doc/api/python/exceptions/exceptions.md [hunk 3]
```diff
@@ -1,33 +1,33 @@
...
+| [InvalidOperationError](./doc/api/python/exceptions/invalidoperationerror.md)   |  InvalidOperationError is raised when a method call is invalid for the object's current state.|
+| [InvalidSelectedItemError](./doc/api/python/exceptions/invalidselecteditemerror.md)   | InvalidSelectedItemError is raised when a method call is valid for the object, but the set value is invalid for the object's current value. Eg: can not select an invalid item for select item method.|
+| [WindowError](./doc/api/python/exceptions/windowerror.md)   | WindowError is raised for common window exceptions.|
+| [WindowsNativeError](./doc/api/python/exceptions/windowsnativeerror.md)   | WindowsNativeError is raised when a method call with win32 native has any exceptions.|
+| [UiaPatternNotFoundError](./doc/api/python/exceptions/uiapatternnotfounderror.md)   | UiaPatternNotFoundError is raised when a desktop UiElement with uia technology can not be found.|
+| [ScreenOperationError](./doc/api/python/exceptions/screenoperationerror.md)   | ScreenOperationError is raised when screen operations with desktop Uielement have any exceptions.|
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/exceptions.md:26]: UiElements

File: doc/api/python/exceptions/notsupportedoperationoptionerror.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 - extends: [NotSupportedOperationError](./doc/api/python/exceptions/notsupportedoperationerror.md)
 
-**NotSupportedOperationOptionError is raised when an invoked method is supported for the target, but the specified option value is not supported. ex: can not clear text for desktop uielement with clear method option set to `ControlClearValue`.**
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/notsupportedoperationoptionerror.md:5]: desktop/window uielement -> desktop UiElement 都统一为后面吧，然后也要和exceptions.md里描述部分保持一致

File: doc/api/python/exceptions/notsupportedoperationoptionerror.md [hunk 5]
```diff
@@ -27,19 +27,19 @@ Stack of the exception which got raised inside the exception.
 ### automation_tech
 - type: str
 
-Type of automation recorder technology. The supported technologies are "uia", "ia", "java", "ie", "chrome", "firefox", "edge" and "sap".
+Type of automation recorder technology. The supporting recorder technologies are as follows "Uia", "Ia", "Java", "IE", "Chrome", "Firefox", "Edge" and "Sap".
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/notsupportedoperationoptionerror.md:30]: 同上

File: doc/api/python/exceptions/notsupportedoperationoptionerror.md [hunk 6]
```diff
@@ -27,19 +27,19 @@ Stack of the exception which got raised inside the exception.
...
 - type: str
 
-Type of automation recorder technology. The supported technologies are "uia", "ia", "java", "ie", "chrome", "firefox", "edge" and "sap".
+Type of automation recorder technology. The supporting recorder technologies are as follows "Uia", "Ia", "Java", "IE", "Chrome", "Firefox", "Edge" and "Sap".
 
 ### control_type
 - type: str
 
-Control type of target element. ex: "Button", "Edit", "Document", "CheckBox", "Image", and the "Image" is specified for image uielment.
+Control type of target element. Eg: "button", "edit", "document", "checkBox", "image", and the "image" is specified for image UiElement.
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/notsupportedoperationoptionerror.md:35]: 同上

File: doc/api/python/exceptions/uiapatternnotfounderror.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 - extends: [WindowError](./doc/api/python/exceptions/windowerror.md)
 
-**UiaPatternNotFoundError is raised when a window uielement with uia technology can not be found.**
+**UiaPatternNotFoundError is raised when a window UiElement with Uia technology can not be found.**
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/uiapatternnotfounderror.md:5]: UIA

File: doc/api/python/exceptions/screenoperationerror.md [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 - extends: [WindowsNativeError](./doc/api/python/exceptions/windowsnativeerror.md)
 
-**ScreenOperationError is raised when screen operations with desktop uielement has any exceptions.**
+**ScreenOperationError is raised when screen operations with desktop UiElement have any exceptions.**
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/screenoperationerror.md:5]: UiElement has

File: doc/api/python/exceptions/notsupportedoperationoptionerror.md (hunk 6 above)
Comment (DiceyClickcorp) [reply doc/api/python/exceptions/notsupportedoperationoptionerror.md:35]: 这里的控件名还是用原来大写的吧。其他同。 “Button", "Edit", "Document", "CheckBox", "Image"

File: doc/api/python/exceptions/exceptions.md [hunk 9]
```diff
@@ -1,33 +1,33 @@
...
-| [ScreenOperationError](./doc/api/python/exceptions/screenoperationerror.md)   | ScreenOperationError is raised when screen operations with desktop uielement has any exceptions|
-| [WebError](./doc/api/python/exceptions/weberror.md) | WebError is raised for common web automation exceptions|
-| [BrowserNotRunError](./doc/api/python/exceptions/browsernotrunerror.md) | BrowserNotRunError is raised when the specified browser is not run|
-| [BrowserNotInstallError](./doc/api/python/exceptions/browsernotinstallerror.md) | BrowserNotInstallError is raised when the specified browser is not installed|
-| [UnreachableBrowserExtensionError](./doc/api/python/exceptions/unreachablebrowserextensionerror.md) | UnreachableBrowserExtensionError is raised when the specified browser's extension not in ready state. May be the extension is not installed, not enabled or failed to run|
-| [WebElementNotRespondingError](./doc/api/python/exceptions/webelementnotrespondingerror.md) | WebElementNotRespondingError is raised when browser's visiting page is not responding|
+| [BaseError](./doc/api/python/exceptions/baseerror.md) |All Clicknium exceptions inherit from this class.|
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/exceptions.md:10]: are inherited?

File: doc/api/python/exceptions/notsupportedoperationerror.md [hunk 10]
```diff
@@ -26,14 +26,14 @@ Stack of the exception which got raised inside the exception.
...
 - type: str
 
-Type of automation recorder technology. The supported technologies are "uia", "ia", "java", "ie", "chrome", "firefox", "edge" and "sap".
+Type of automation recorder technology. The supporting technologies are as follows,"UIA", "IA", "Java", "Ie", "Chrome", "Firefox", "Edge" and "SAP".
 
 ### control_type
 - type: str
 
-Control type of target element. ex: "Button", "Edit", "Document", "CheckBox", "Image", and the "Image" is specified for image uielment.
+Control type of target element. Eg: "button", "edit", "document", "checkBox", "image", and the "image" is specified for image UiElment.
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/notsupportedoperationerror.md:34]: 这里的控件名还是用原来大写的吧。其他同。 “Button", "Edit", "Document", "CheckBox", "Image"

File: doc/api/python/exceptions/argumenterror.md [hunk 11]
```diff
@@ -2,7 +2,7 @@
 
 - extends: [BaseError](./doc/api/python/exceptions/baseerror.md)
 
-**ArgumentError is raised when one of the arguments provided to a method is not valid.**
+**ArgumentError is raised when at least one of the arguments provided to a method is not valid.**
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/argumenterror.md:5]: 这里的描述，要和exceptions.md里的描述要保持一致。同样exceptions.md改的话，也要修改对应的error.md

File: doc/api/python/exceptions/exceptions.md [hunk 12]
```diff
@@ -1,33 +1,33 @@
...
+| [ArgumentError](./doc/api/python/exceptions/argumenterror.md) |  ArgumentError is raised when one of the arguments provided to a method is not valid.|
+| [ArgumentNullError](./doc/api/python/exceptions/argumentnullerror.md) | ArgumentNullError is raised when a null reference is passed to a method that does not accept it as a valid argument.|
+| [ArgumentOutOfRangeError](./doc/api/python/exceptions/argumentoutofrangeerror.md) | ArgumentOutOfRangeError is raised when the value of an argument is out of the allowable range of values as defined by the invoked method.|
+| [LocatorUndefinedError](./doc/api/python/exceptions/locatorundefinederror.md) | LocatorUndefinedError is raised when the specified locator can not be found in locator store.|
+| [TimeoutOperationError](./doc/api/python/exceptions/timeoutoperationerror.md) | TimeoutOperationError is raised when hen a certain operation is not completed within given time.|
```
Comment (robinbob) [inline doc/api/python/exceptions/exceptions.md:15]: OperationTimeoutError  @DiceyClickcorp

File: doc/api/python/exceptions/exceptions.md [hunk 13]
```diff
@@ -1,33 +1,33 @@
...
+| [ElementCanNotFoundError](./doc/api/python/exceptions/elementcannotfounderror.md) | ElementCanNotFoundError is raised when the loactor can be found in locator store, but the returned UiElement can not be found.|
```
Comment (robinbob) [inline doc/api/python/exceptions/exceptions.md:16]: ElementNotFoundError

File: doc/api/python/exceptions/exceptions.md (hunk 12 above)
Comment (DiceyClickcorp) [reply doc/api/python/exceptions/exceptions.md:15]: will fixed with code

File: doc/api/python/exceptions/exceptions.md (hunk 13 above)
Comment (DiceyClickcorp) [reply doc/api/python/exceptions/exceptions.md:16]: will fixed with code

File: doc/api/python/exceptions/screenoperationerror.md (hunk 8 above)
Comment (Luluyu2022) [reply doc/api/python/exceptions/screenoperationerror.md:5]: screen operations with ...应该要have...

File: doc/api/python/exceptions/exceptions.md (hunk 9 above)
Comment (Luluyu2022) [reply doc/api/python/exceptions/exceptions.md:10]: exceptions 应该是主语，直接从级别里继承，应该直接inherit更合适。

File: doc/api/python/exceptions/exceptions.md [hunk 14]
```diff
@@ -1,33 +1,33 @@
...
+| [ScreenOperationError](./doc/api/python/exceptions/screenoperationerror.md)   | ScreenOperationError is raised when screen operations with desktop UiElements have any exceptions.|
+| [WebError](./doc/api/python/exceptions/weberror.md) | WebError is raised for common web automation exceptions.|
+| [BrowserNotRunError](./doc/api/python/exceptions/browsernotrunerror.md) | BrowserNotRunError is raised when the specified browser is not running.|
+| [BrowserNotInstallError](./doc/api/python/exceptions/browsernotinstallerror.md) | BrowserNotInstallError is raised when the specified browser is not installed.|
+| [UnreachableBrowserExtensionError](./doc/api/python/exceptions/unreachablebrowserextensionerror.md) | UnreachableBrowserExtensionError is raised when the specified browser's extension not in ready state. Maybe the extension is not installed, not enabled or failed to run.|
+| [WebElementNotRespondingError](./doc/api/python/exceptions/webelementnotrespondingerror.md) | WebElementNotRespondingError is raised when the web page is not responding.|
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/exceptions.md:31]: this change should also be applied to file webelementnotrespondingerror.md.

File: doc/api/python/exceptions/notsupportedoperationoptionerror.md [hunk 15]
```diff
@@ -32,12 +32,12 @@ Type of automation recorder technology. The supporting recorder technologies are
...
 - type: str
 
-Control type of target element. Eg: "button", "edit", "document", "checkBox", "image", and the "image" is specified for image UiElement.
+Control type of target element. Eg: "Button", "Edit", "Document", "CheckBox", "Image", and the "Image" is specified for image UiElement.
 
 ### operation
 - type: str
 
-Operation methods of the target element. Eg: "click", "check", "highlight".
+Operation methods of the target element. Eg: "Click", "Check", "Highlight".
```
Comment (DiceyClickcorp) [inline doc/api/python/exceptions/notsupportedoperationoptionerror.md:40]: "click", "check", "highlight" -> not need to uppercase here.

Comment (Luluyu2022) [reply doc/api/python/exceptions/notsupportedoperationoptionerror.md:40]: @DiceyClickcorp could you help me confirm which operation needs to be capitalized?
I am confused about this.

Comment (DiceyClickcorp) [reply doc/api/python/exceptions/notsupportedoperationoptionerror.md]: for methods , should be lowercase. for control type, should be uppercase

**Review by DiceyClickcorp — APPROVED** [review]

**Pull request merged**